src/Details.js

```diff
@@ -41,7 +21,24 @@ function DetailsBase({children, overlay, render = getRenderer(children), ...rest
 
   function toggle(event) {
     if (event) event.preventDefault()
-    setOpen(!open)
+    if (overlay) {
+      openMenu()
+    } else {
+      setOpen(!open)
+    }
+  }
+
+  function openMenu() {
+    if (!open) {
+      setOpen(true)
+      document.addEventListener('click', closeMenu)
```

Comment:
I don't really understand the premise here 😬. From the original issue body it sounds like those menus just shouldn't use the optional `overlay`?

But looks like this will also close details when user clicks inside the menu which wasn't the case.

Reply:
The reason why I didn't want to use the CSS overlay is that when there is a menu open, and I want to click on another menu to open the new menu, clicking only closes the first menu, and doesn't open the second one (the second menu's click never triggers because it's behind the overlay)

Reply:
And yeah, we do want to close the menu when the user clicks inside of a menu because that will jump them down to a different part of the page and at that point the menu can/should be closed

Reply:
Let me see if I can get deploys working so that y'all can play around with it

Reply:
In dotcom there is a JS class that listens to `<details>`'s toggle open event, and close other `<details open>` with the same class name; it can be used with or without `.details-overlay`. 

When with `.details-overlay`, the only way to get into that state is with keyboard, which was the use case the class was initially made for.